src/main/java/io/jsonwebtoken/impl/crypto/ECDSA.java

```diff
@@ -0,0 +1,177 @@
+package io.jsonwebtoken.impl.crypto;
+
+
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.lang.JOSEException;
+
+
+/**
+ * Elliptic Curve Digital Signature Algorithm (ECDSA) functions and utilities.
+ */
+public class ECDSA {
```

Comment:
We only accept 'utils' classes for things the JDK makes final (e.g. StringUtils).  This logic should probably go in the EllipticCurveProvider class?